src/pages/index.js

```diff
@@ -1,101 +1,245 @@
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "eac0c1a2-60c5-44c4-979d-14695d75b6b5",
+    "Content-Type": "application/json",
+  },
+});
 
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: renderCard,
   },
   ".cards__list"
 );
 section.renderItems();
 
-const popupEditForm = new PopupWithForm(
+api
+  .getInitialCards()
+  .then((cards) => {
+    section.setItems(cards);
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+/* -------------------------------------------------------------------------- */
+/*                                  Popups                                    */
+/* -------------------------------------------------------------------------- */
+const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
-popupEditForm.setEventListeners();
+editProfilePopup.setEventListeners();
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+editAvatarPopup.setEventListeners();
 
-const popupAddForm = new PopupWithForm(
+const addCardPopup = new PopupWithForm(
   "#profile-add-modal",
   handleAddCardSubmit
 );
-popupAddForm.setEventListeners();
+addCardPopup.setEventListeners();
 
 const popupImage = new PopupWithImage("#modal__image");
-popupImage.setEventListiners();
+popupImage.setEventListeners();
 
 const userInfo = new UserInfo({
   titleSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatar: ".profile__image",
 });
 
 // FUNCTIONS/////
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getCardElement();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleLikeClicks,
+    handleDeleteSubmit
+  );
+  return card.getView();
 }
 
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
 
-function handleImageClick(name, link) {
-  popupImage.open(name, link);
+function handleImageClick(cardData) {
+  popupImage.open(cardData);
 }
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  popupEditForm.close();
+function handleAvatarSubmit(url) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(res);
+      avatarForm.reset();
+      avatarFormValidator.resetValidation();
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => editAvatarPopup.renderLoading(false));
 }
 
-function handleAddCardSubmit(cardData) {
-  renderCard({ name: cardData.title, link: cardData.url });
-  cardForm.reset();
-  addFormValidator.disableButton();
-  popupEditForm.close();
+function handleProfileEditSubmit({ name, about }) {
+  editProfilePopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```